### DISCUSSION TITLE
Transformations during replication

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#14570] [General]
Hi,

I have scenario where I want to replicate production data to a sandbox database, but would like to apply some transformations to the data to anonymise it during replication.

Is this possible and if not what which approach you recommend for this?

Thanks

**Comment by ayende** [comment]
Certainly, you can do that using the RavenDB ETL. It is exactly for that scenario.

**Reply by nickchampion** [reply]
Ah yes, thanks @ayende I had one question on the on-going tasks, it is possible & documented anywhere to set up ETL processes via the Node.JS client, I want to be able to automate the process and deploy the scripts as part of our deployment process?

Thanks

**Reply by ayende** [reply]
Yes, you can do that. See here for details:
https://github.com/ravendb/ravendb-nodejs-client/blob/v5.2/test/Ported/Server/Etl/Raven/EtlTest.ts#L33
